### PR TITLE
Subscription renewal orders are missing the statement descriptor in Stripe

### LINKED ISSUE #3450: Statement descriptor missing from renewal orders
**Describe the bug**

Slack report: p1726912653375519-slack-C7U3Y3VMY

Subscription renewal orders are missing statement descriptor information:

image

As of Jan 2024, using the `statement_descriptor` parameter for card payments has been deprecated and we're suppose to be using `statement_descriptor_suffix` instead but it looks like our renewal requests aren't using the new param yet.

**To Reproduce**
1. In your stripe settings, enable the "Add customer order number to the bank statement" setting
2. Purchase a subscription product with Stripe
3. Go to https://dashboard.stripe.com/test/payments and view the parent order and confirm the order number in statement descriptor is present:
image
4. Process a renewal order and view the latest charge in stripe and confirm no `statement_descriptor_suffix` param

Related: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2833

### PR DESCRIPTION
**Pull request by mattallan**
Fixes #3450

## Changes proposed in this Pull Request:

This PR addresses the issue where subscription renewal orders are missing the `statement_descriptor_suffix` parameter, which is necessary after the deprecation of the `statement_descriptor` parameter in January 2024.

| Parent | Renewal :x: |
|--------|--------|
| image|image| 

In https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2833 we updated all of our code to use `statement_descriptor_suffix` instead of the deprecated `statement_descriptor` but I believe the `create_and_confirm_intent_for_off_session` function was missed in the PR, which impacts renewal orders.

## Testing instructions

1. In your stripe settings, enable the "Add customer order number to the bank statement" setting
2. Purchase a subscription product with Stripe
3. Go to https://dashboard.stripe.com/test/payments and view the parent order and confirm the order number in statement descriptor is present:
image
4. Process a renewal order and view the latest charge in stripe and confirm the `statement_descriptor_suffix` param is now populated while on this branch

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

**Review by james-allan — APPROVED** [review]
Changes look good. I confirmed that on `develop` that the statement descriptor was missing the order ID and its fixed on this branch (see screenshots below).

| `develop` | This branch |
|--------|--------|
| Screenshot 2024-09-24 at 3 03 42 pm | Screenshot 2024-09-24 at 3 04 45 pm | 

I also double checked that that with the order number setting disabled, we don't the order number in the descriptor.

File: changelog.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@
 * Tweak - Update the Apple Pay domain registration flow to use the new Stripe API endpoint.
 * Fix - Fix empty error message for Express Payments when order creation fails.
 * Fix - Fix multiple issues related to the reuse of Cash App Pay tokens (as a saved payment method) when subscribing.
+* Fix - Add missing statement descriptor in Stripe for subscription renewal orders.
```
Comment (james-allan) [inline changelog.txt]: I'm not sure if we want to clarify that its the order number that was missing. This changelog makes it sound as if no statement descriptor was included.

Comment (james-allan) [inline changelog.txt]: eg 
```suggestion
* Fix - Ensure subscription renewal order statement descriptors include the order number suffix.
```

Idk something like that. Feel free to use that or another option. :)

**Comment by mattallan** [comment]
Thanks for the review @james-allan, I've copied the updated changelog to readme.txt.

I'll merge this once the checks pass.

File: changelog.txt (hunk 1 above)
Comment (mattallan) [reply changelog.txt]: Fixed in [c1724ab](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3451/commits/c1724ab338cae9592fb1c4bb557a546150b4978a) and [2f82887](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3451/commits/2f82887a339ddf9b3f98e875f380bed889f0041d)

**Pull request merged**